osu.Server.Spectator/Services/SharedInterop.cs

```diff
@@ -178,16 +178,8 @@ private class RoomWithHostId : Room
             /// Creates a <see cref="Room"/> from a <see cref="MultiplayerRoom"/>.
             /// </summary>
             public RoomWithHostId(MultiplayerRoom room)
+                : base(room)
             {
-                RoomID = room.RoomID;
-                Host = room.Host?.User;
```

Comment:
This is technically different than [what the base ctor that's being called in the new code does](https://github.com/ppy/osu/blob/bd590786710afeff63c40f65361b290d42f61e2a/osu.Game/Online/Rooms/Room.cs#L358). Did you test that this doesn't cause any weird issues? I'd like to think that it doesn't but this is the sort of nightmare model juggling stuff that breaks more often than I'd want to see.

Reply:
Yeah, this is fine.

First, this value is [overwritten](https://github.com/ppy/osu-web/blob/0e49d567e73da910d9340cd08b87143ce8efc570/app/Models/Multiplayer/Room.php#L608) by osu!web. 

Second, `.User` should actually always be null in spectator because it's only used osu-side as a post-processing variable, so the base ctor is more correct in this case, though it doesn't matter because of the above. Now that I say it out loud, the `.User` property should be removed from this model altogether, but that's for another time.